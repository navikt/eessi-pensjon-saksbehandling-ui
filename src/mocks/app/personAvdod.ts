export default () => {
  const x = Math.random()
  const whichOne = Math.floor(x * 3)
  if (whichOne === 0) {
    return []
  }
  if (whichOne === 1) {
    return [{
      fnr: '12345678901'
    }]
  }
  if (whichOne === 2) {
    return [{
      fnr: '12345678902'
    }, {
      fnr: '12345678903'
    }]
  }
  return undefined
}
