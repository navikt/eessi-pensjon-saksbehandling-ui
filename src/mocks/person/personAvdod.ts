const avdod1 = {
  fnr: 'personFarFnr',
  aktoerId: '2865434801175',
  fulltNavn: 'MOR AVDØD1',
  fornavn: 'MOR',
  mellomnavn: null,
  etternavn: 'AVDØD1',
  relasjon: 'MOR'
}

const avdod2 = {
  fnr: 'personMorFnr',
  aktoerId: '2865434801175',
  fulltNavn: 'FAR AVDØD2',
  fornavn: 'FAR',
  mellomnavn: null,
  etternavn: 'AVDØD2',
  relasjon: 'FAR'
}

export default (nrAvdod: number | undefined) => {
  let whichOne = 0
  if (nrAvdod !== undefined) {
    whichOne = nrAvdod
    if (whichOne > 2) {
      whichOne = 2
    }
  } else {
    whichOne = Math.floor(Math.random() * 3)
  }
  if (whichOne === 0) {
    return []
  }
  if (whichOne === 1) {
    return [avdod1]
  }
  if (whichOne === 2) {
    return [avdod1, avdod2]
  }
  return undefined
}
