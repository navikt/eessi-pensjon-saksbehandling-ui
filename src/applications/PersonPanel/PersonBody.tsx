import { PersonIcon, HouseIcon, CalendarIcon, GlobeIcon, PersonRectangleIcon, HeartIcon } from '@navikt/aksel-icons'
import PostalCodes from 'src/components/PostalCodes/PostalCodes'
import { AllowedLocaleStringPropType } from 'src/declarations/app.pt'
import { PersonAvdod, PersonAvdods } from 'src/declarations/person.d'
import CountryData from '@navikt/land-verktoy'
import _ from 'lodash'
import moment from 'moment'
import { BodyLong, Box, Detail, HStack } from '@navikt/ds-react'
import PT from 'prop-types'
import { useTranslation } from 'react-i18next'
import styled from 'styled-components'
import { HorizontalLineSeparator, HorizontalSeparatorSpan } from 'src/components/StyledComponents'
import PopoverCustomized from "src/components/Tooltip/PopoverCustomized";

const Element = styled.div`
  display: flex;
  align-items: baseline;
`
const MarginRow = styled(HStack)`
  margin: 1.5rem;
  display: flex;
`
const MarginColumn = styled.div`
  display: flex;
  align-items: flex-start;
  margin-bottom: 0.5rem;
  flex: ${(props: any) => props.flex || 1};
  margin-left: 0.5rem;
  margin-right: 0.5rem;
`
export const PersonBodyDiv = styled.div``

export interface PersonBodyProps {
  locale: string
  person: any
  personAvdods: PersonAvdods | undefined
}

const PersonBody: React.FC<PersonBodyProps> = ({
  locale,
  person,
  personAvdods
}: PersonBodyProps): JSX.Element | null => {
  const { t } = useTranslation()

  const renderEntity = (label: string, value: any): JSX.Element => {
    let _value
    if (!value) {
      _value = [t('ui:notRegistered')]
    } else {
      _value = typeof value === 'string' ? [value] : value
    }
    if (_.isEmpty(_value)) {
      _value = [t('ui:notRegistered')]
    }
    return (
      <Element
        id={'w-overview-PersonBody--element-' + label.replace('ui:', '')}
        data-testid={'w-overview-PersonBody--element-' + label.replace('ui:', '')}
      >
        <Detail>
          <strong>
            {t(label)}
          </strong>:
        </Detail>
        <Box paddingInline="4 0">
          <div className='navds-body-long'>
            {_value.map((val: any) => val)}
          </div>
        </Box>
      </Element>
    )
  }

  const addAddressLine = (
    address: Array<JSX.Element | string>,
    value: string | null,
    label: string,
    separator: JSX.Element
  ): Array<JSX.Element | string> => {
    if (value) {
      address.push((
        <PopoverCustomized label={<span>{label}</span>} key={"tooltip-" + + label + value}>
          <span key={value}>{value}</span>
        </PopoverCustomized>
      ))
      address.push(separator)
    }
    return address
  }

  const getCountry = (value: string): string | null => {
    const nationality = CountryData.getCountryInstance(locale).findByValue3(value)
    return nationality ? nationality.label : null
  }

  const getDates = (person: any, category: string): string => {
    let beginDate = _.get(person, category + '.gyldigFraOgMed')
    if (beginDate) {
      beginDate = moment(beginDate).format('DD.MM.YYYY')
    }
    let endDate = ''
    if (_.get(person, category + '.gyldigTilOgMed')) {
      endDate = _.get(person, category + '.gyldigTilOgMed')
      if (endDate) {
        endDate = moment(endDate).format('DD.MM.YYYY')
      }
    }
    return beginDate + ' - ' + endDate
  }

  const getAddress = (person: any, category: string): Array<JSX.Element | string> => {
    if (!_.get(person, category + '.vegadresse')) {
      return []
    }
    let adresse: Array<JSX.Element | string> = []
    const husnummer = _.get(person, category + '.vegadresse.husnummer')
    const husbokstav = _.get(person, category + '.vegadresse.husbokstav')
    adresse = addAddressLine(adresse, getDates(person, category), t('ui:fram-og-til'), <br key={0} />)
    adresse = addAddressLine(adresse, _.get(person, category + '.vegadresse.adressenavn'), t('ui:adressenavn'),
      husnummer || husbokstav ? <HorizontalSeparatorSpan key={1} /> : <br key={1} />)
    adresse = addAddressLine(adresse, husnummer, t('ui:husnummer'), husbokstav ? <HorizontalSeparatorSpan key={2} /> : <br key={2} />)
    adresse = addAddressLine(adresse, husbokstav, t('ui:husbokstav'), <br key={3} />)
    const zipCode = _.get(person, category + '.vegadresse.postnummer')
    if (zipCode) {
      adresse = addAddressLine(adresse, zipCode, t('ui:poststed'), <HorizontalSeparatorSpan key={4} />)
      adresse = addAddressLine(adresse, PostalCodes.get(zipCode), t('ui:city'), <br key={5} />)
    }
    return adresse
  }

  if (!person) {
    return null
  }

  let birthDateString
  let deathDateString
  let nationality: Array<string> = []
  let maritalStatus: Array<string> = []
  const bostedsadresse: Array<JSX.Element | string> = getAddress(person, 'bostedsadresse')
  const oppholdsadresse: Array<JSX.Element | string> = getAddress(person, 'oppholdsadresse')

  if (_.get(person, 'foedsel.foedselsdato')) {
    birthDateString = moment(person.foedsel.foedselsdato).format('DD.MM.YYYY')
  }
  if (_.get(person, 'doedsfall.doedsdato')) {
    deathDateString = moment(person.doedsfall.doedsdato).format('DD.MM.YYYY')
  }
  if (!_.isEmpty(person.statsborgerskap)) {
    nationality = person.statsborgerskap.map((l: any) => {
      let label = getCountry(l.land)
      if (l.gyldigFraOgMed) {
        label += ' (' + moment(l.gyldigFraOgMed).format('DD.MM.YYYY')
      }
      if (l.gyldigTilOgMed) {
        label += ' - ' + moment(l.gyldigTilOgMed).format('DD.MM.YYYY')
      }
      return label + ')'
    })
  }

  if (!_.isEmpty(person.sivilstand)) {
    maritalStatus = person.sivilstand.map((s: any) => {
      let type = s.type.toLowerCase()
      type = type.charAt(0).toUpperCase() + type.slice(1)
      let dateString = ''
      if (s.gyldigFraOgMed) {
        dateString = moment(s.gyldigFraOgMed).format('DD.MM.YYYY')
      }
      if (s.gyldigTilOgMed) {
        dateString += ' - ' + moment(s.gyldigTilOgMed).format('DD.MM.YYYY')
      }
      const label = t(type) +
        (type !== 'Null' && type !== 'Ugift' && dateString
          ? ' (' + dateString + ')'
          : '')
      return label
    })
  }

  return (
    <PersonBodyDiv data-testid="person-body-div">
      <MarginRow key="personBody-marginrow-1">
        <MarginColumn key="personBody-marginrow-1-margincolumn-1">
          <HStack gap="4">
            <HouseIcon title={t('ui:bostedsadresse')} fontSize="1.5rem" />
            {bostedsadresse ? renderEntity('ui:bostedsadresse', bostedsadresse) : null}
          </HStack>
        </MarginColumn>
        <MarginColumn key="personBody-marginrow-1-margincolumn-2">
          <HStack gap="4">
            <CalendarIcon title={t('ui:birthdate')} fontSize="1.5rem" />
            {renderEntity('ui:birthdate', birthDateString)}
          </HStack>
        </MarginColumn>
        <MarginColumn key="personBody-marginrow-1-margincolumn-3">
          <HStack gap="4">
            <PersonRectangleIcon title={t('ui:nationality')} fontSize="1.5rem"/>
            {renderEntity('ui:nationality', nationality)}
          </HStack>
        </MarginColumn>
      </MarginRow>
      <HorizontalLineSeparator />
      <MarginRow key="personBody-marginrow-2">
        <MarginColumn key="personBody-marginrow-2-margincolumn-1">
          <HStack gap="4">
            <GlobeIcon title={t('ui:oppholdsadresse')} fontSize="1.5rem" />
            {renderEntity('ui:oppholdsadresse', oppholdsadresse)}
          </HStack>
        </MarginColumn>
        <MarginColumn key="personBody-marginrow-2-margincolumn-2">
          <HStack gap="4">
            <HeartIcon title={t('ui:marital-status')} fontSize="1.5rem"/>
            {renderEntity('ui:marital-status', maritalStatus)}
          </HStack>
        </MarginColumn>
        <MarginColumn key="personBody-marginrow-2-margincolumn-3"/>
      </MarginRow>
      {((personAvdods && personAvdods.length > 0) || deathDateString) && (
        <>
          <HorizontalLineSeparator />
          <MarginRow key="personBody-marginrow-3">

            <MarginColumn key="personBody-marginrow-3-margincolumn-1">
              <PersonIcon fontSize="1.5rem" />
              <Box paddingInline="4 0">
                <Detail size='small'>
                  <strong>{t('ui:deceased')}</strong>:
                </Detail>
               </Box>
              <div>
                {personAvdods && personAvdods.length > 0
                  ? personAvdods.map((avdod: PersonAvdod) => (
                    <Element
                      key={avdod.fnr}
                      id='w-overview-PersonBody--element-deceased'
                    >
                      <Box paddingInline="4 0">
                        <BodyLong>
                          {avdod?.fornavn +
                            (avdod?.mellomnavn ? ' ' + avdod?.mellomnavn : '') +
                            (avdod?.etternavn ? ' ' + avdod?.etternavn : '') +
                            ' - ' + avdod.fnr}
                          {avdod.relasjon ? ' (' + t('buc:relasjon-' + avdod?.relasjon) + ')' : ''}
                          {avdod?.doedsDato ? ' (DÃ¸dsdato: ' + avdod.doedsDato + ')' : ''}

                        </BodyLong>
                      </Box>
                    </Element>
                  ))
                  : (
                    <Element
                        key='noAvdod'
                        id='w-overview-PersonBody--element-deceased'
                      >
                        <Box paddingInline="4 0">
                          <BodyLong>
                            {t('buc:form-noAvdod')}
                          </BodyLong>
                        </Box>
                      </Element>
                      )}
              </div>
            </MarginColumn>
            {deathDateString
              ? (
                <MarginColumn key="personBody-marginrow-3-margincolumn-2">
                  <HStack gap="4">
                    <CalendarIcon fontSize="1.5rem" />
                    {renderEntity('ui:deathdate', deathDateString)}
                  </HStack>
                </MarginColumn>
                )
              : (
                <MarginColumn key="personBody-marginrow-3-margincolumn-2"/>
                )}
          </MarginRow>
        </>
      )}
    </PersonBodyDiv>
  )
}

PersonBody.propTypes = {
  locale: AllowedLocaleStringPropType.isRequired,
  person: PT.object
}

export default PersonBody
