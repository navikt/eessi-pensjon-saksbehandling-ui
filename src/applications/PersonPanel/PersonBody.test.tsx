import { screen, render } from '@testing-library/react'
import personAvdod from 'src/mocks/person/personAvdod'
import mockPerson from 'src/mocks/person/personPdl'
import PersonBody, { PersonBodyProps } from './PersonBody'

describe('applications/PersonPanel/PersonBody', () => {
  const initialMockProps: PersonBodyProps = {
    locale: 'nb',
    person: mockPerson,
    personAvdods: personAvdod(1)
  }

  const mockPropsEmptyValue: PersonBodyProps = {
    locale: 'nb',
    person: {
      ...mockPerson,
      bostedsadresse: null,
      foedselsdato: {}
    },
    personAvdods: personAvdod(1)
  }

  it('Render: has proper HTML structure', () => {
    render(<PersonBody {...initialMockProps} />)

    expect(screen.getByTestId("person-body-div")).toBeInTheDocument()
    expect(screen.getByTitle("ui:bostedsadresse")).toBeInTheDocument()
    expect(screen.getByTitle("ui:birthdate")).toBeInTheDocument()
    expect(screen.getByTitle("ui:nationality")).toBeInTheDocument()
    expect(screen.getByTitle("ui:oppholdsadresse")).toBeInTheDocument()
    expect(screen.getByTitle("ui:marital-status")).toBeInTheDocument()
    expect(screen.getByTestId("w-overview-PersonBody--element-bostedsadresse")).toHaveTextContent(/ui:bostedsadresse:ui:fram-og-til01.01.2020 - 01.01.2021ui:adressenavnAdressenavnui:husnummer00ui:husbokstavAui:poststed0768ui:cityOSLO/i)
    expect(screen.getByTestId("w-overview-PersonBody--element-birthdate")).toHaveTextContent(/ui:birthdate:09.02.1980/i)
    expect(screen.getByTestId("w-overview-PersonBody--element-nationality")).toHaveTextContent("ui:nationality:DanmarkNorge (09.02.1980 - 09.02.1981)Belgia")
  })

  it('Render: Empty value renders not registered', () => {
    render(<PersonBody {...mockPropsEmptyValue} />)
    expect(screen.getByTestId("person-body-div")).toBeInTheDocument()
    expect(screen.getByTitle("ui:bostedsadresse")).toBeInTheDocument()
    expect(screen.getByTestId("w-overview-PersonBody--element-bostedsadresse")).toHaveTextContent(/ui:bostedsadresse:ui:notRegistered/i)
    expect(screen.getByTestId("w-overview-PersonBody--element-birthdate")).toHaveTextContent(/ui:birthdate:ui:notRegistered/i)
  })

  it('Render: gets dates converted properly', () => {
    render(<PersonBody {...initialMockProps} />)
    expect(screen.getByTestId("w-overview-PersonBody--element-marital-status")).toHaveTextContent(/ui:marital-status:Gift \(10.10.2007\)/i)
  })
})
