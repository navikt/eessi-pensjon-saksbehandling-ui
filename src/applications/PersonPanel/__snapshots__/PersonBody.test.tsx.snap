// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applications/PersonPanel/PersonBody Render: match snapshot 1`] = `
.c13 {
  margin: 0.3rem;
  color: var(--navds-semantic-color-text);
}

ui-fss.g1[id="c12"] {
  content: "c13,";
}

.c15 {
  display: inline-block;
}

ui-fss.g2[id="c14"] {
  content: "c15,";
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: -0.5rem;
  margin-right: -0.5rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

ui-fss.g10[id="c0"] {
  content: "c2,";
}

.c6 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-left: 0.5rem;
  margin-right: 0.5rem;
}

ui-fss.g14[id="c4"] {
  content: "c6,";
}

.c17 {
  display: inline-block;
  margin-left: 1rem;
}

ui-fss.g19[id="c16"] {
  content: "c17,";
}

.c9 {
  display: inline-block;
  margin-left: 1rem;
}

ui-fss.g20[id="c8"] {
  content: "c9,";
}

.c19 {
  height: 1px;
  background: linear-gradient(90deg,var(--navds-semantic-color-canvas-background) 0%,var(--navds-semantic-color-border) 5%,var(--navds-semantic-color-border) 95%,var(--navds-semantic-color-canvas-background) 100% );
  width: -webkit-fill-available;
  margin-left: 2rem;
  margin-right: 2rem;
}

ui-fss.g45[id="c18"] {
  content: "c19,";
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

ui-fss.g46[id="c10"] {
  content: "c11,";
}

.c3 {
  margin: 1.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

ui-fss.g47[id="c1"] {
  content: "c3,";
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  margin-bottom: 0.5rem;
}

ui-fss.g48[id="c5"] {
  content: "c7,";
}

@media (min-width:768px) {
  .c2 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<div
  class=""
>
  <div
    class="c0 c1 c2 c3"
  >
    <div
      class="c4 c5 c6 c7"
    >
      <svg
        fill="none"
        focusable="false"
        height="1em"
        role="img"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="m12 0 10 10v14h-9v-6h-2v6H2V10L12 0Zm0 2.829-8 7.999V22h5v-6h6v6h5V10.829l-8-8Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      <div
        class="c8 c9"
      />
      <div
        class="c10 c11"
        id="w-overview-PersonBody--element-bostedsadresse"
      >
        <p
          class="navds-detail"
        >
          <strong>
            ui:bostedsadresse
          </strong>
          :
        </p>
        <div
          class="c8 c9"
        />
        <div
          class="navds-body-long"
        >
          <div
            aria-hidden="true"
            class="navds-popover navds-popover--hidden"
            data-placement="right"
            style="position: absolute; top: 0px; left: 0px;"
            tabindex="-1"
          >
            <div
              class="c12 c13"
            >
              <span>
                ui:fram-og-til
              </span>
            </div>
            <div
              class="navds-popover__arrow"
              style="left: -0.5rem;"
            />
          </div>
          <div
            class="c14 c15"
            id="dcfe61d9-94d0-467c-bba9-782d8d10eb94"
          >
            <span>
              01.01.2020 - 01.01.2021
            </span>
          </div>
          <br />
          <div
            aria-hidden="true"
            class="navds-popover navds-popover--hidden"
            data-placement="right"
            style="position: absolute; top: 0px; left: 0px;"
            tabindex="-1"
          >
            <div
              class="c12 c13"
            >
              <span>
                ui:adressenavn
              </span>
            </div>
            <div
              class="navds-popover__arrow"
              style="left: -0.5rem;"
            />
          </div>
          <div
            class="c14 c15"
            id="51b24c6d-9c3e-4a48-bcc4-e7198ace21a0"
          >
            <span>
              Adressenavn
            </span>
          </div>
          <span
            class="c16 c17"
          />
          <div
            aria-hidden="true"
            class="navds-popover navds-popover--hidden"
            data-placement="right"
            style="position: absolute; top: 0px; left: 0px;"
            tabindex="-1"
          >
            <div
              class="c12 c13"
            >
              <span>
                ui:husnummer
              </span>
            </div>
            <div
              class="navds-popover__arrow"
              style="left: -0.5rem;"
            />
          </div>
          <div
            class="c14 c15"
            id="fc55ce70-1170-4193-a4f5-842891fe91a8"
          >
            <span>
              00
            </span>
          </div>
          <span
            class="c16 c17"
          />
          <div
            aria-hidden="true"
            class="navds-popover navds-popover--hidden"
            data-placement="right"
            style="position: absolute; top: 0px; left: 0px;"
            tabindex="-1"
          >
            <div
              class="c12 c13"
            >
              <span>
                ui:husbokstav
              </span>
            </div>
            <div
              class="navds-popover__arrow"
              style="left: -0.5rem;"
            />
          </div>
          <div
            class="c14 c15"
            id="f0fadca5-3ed1-4b25-9cf7-76060b11da3c"
          >
            <span>
              A
            </span>
          </div>
          <br />
          <div
            aria-hidden="true"
            class="navds-popover navds-popover--hidden"
            data-placement="right"
            style="position: absolute; top: 0px; left: 0px;"
            tabindex="-1"
          >
            <div
              class="c12 c13"
            >
              <span>
                ui:poststed
              </span>
            </div>
            <div
              class="navds-popover__arrow"
              style="left: -0.5rem;"
            />
          </div>
          <div
            class="c14 c15"
            id="0157d0a9-677f-437c-a110-b83fabd78b4b"
          >
            <span>
              0768
            </span>
          </div>
          <span
            class="c16 c17"
          />
          <div
            aria-hidden="true"
            class="navds-popover navds-popover--hidden"
            data-placement="right"
            style="position: absolute; top: 0px; left: 0px;"
            tabindex="-1"
          >
            <div
              class="c12 c13"
            >
              <span>
                ui:city
              </span>
            </div>
            <div
              class="navds-popover__arrow"
              style="left: -0.5rem;"
            />
          </div>
          <div
            class="c14 c15"
            id="8712f926-9db7-4dbd-bdba-13e294bfe6e3"
          >
            <span>
              OSLO
            </span>
          </div>
          <br />
        </div>
      </div>
    </div>
    <div
      class="c4 c5 c6 c7"
    >
      <svg
        fill="none"
        focusable="false"
        height="1em"
        role="img"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M6 7V5H2v5h20V5h-4v2a1 1 0 1 1-2 0V5H8v2a1 1 0 1 1-2 0Zm10-4H8V1a1 1 0 1 0-2 0v2H2a2 2 0 0 0-2 2v17a2 2 0 0 0 2 2h20a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2h-4V1a1 1 0 1 0-2 0v2ZM2 12v10h20V12H2Zm6 3a1 1 0 0 0-1-1H5a1 1 0 1 0 0 2h2a1 1 0 0 0 1-1Zm-1 3a1 1 0 1 1 0 2H5a1 1 0 1 1 0-2h2Zm6-4h-2a1 1 0 1 0 0 2h2a1 1 0 1 0 0-2Zm-2 4h2a1 1 0 1 1 0 2h-2a1 1 0 1 1 0-2Zm9-3a1 1 0 0 0-1-1h-2a1 1 0 1 0 0 2h2a1 1 0 0 0 1-1Zm-4 4a1 1 0 0 1 1-1h2a1 1 0 1 1 0 2h-2a1 1 0 0 1-1-1Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      <div
        class="c8 c9"
      />
      <div
        class="c10 c11"
        id="w-overview-PersonBody--element-birthdate"
      >
        <p
          class="navds-detail"
        >
          <strong>
            ui:birthdate
          </strong>
          :
        </p>
        <div
          class="c8 c9"
        />
        <div
          class="navds-body-long"
        >
          09.02.1980
        </div>
      </div>
    </div>
    <div
      class="c4 c5 c6 c7"
    >
      <svg
        fill="none"
        focusable="false"
        height="1em"
        role="img"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M19 2H5v20h14V2ZM3 0v24h18V0H3Zm10 7a1 1 0 1 1-2 0 1 1 0 0 1 2 0Zm2 0a3 3 0 1 1-6 0 3 3 0 0 1 6 0Zm-3 3c1.08 0 2.092.401 2.835 1.159.704.717 1.117 1.703 1.161 2.841H16v.205h-2c0-.758-.252-1.299-.593-1.646C13.065 12.21 12.577 12 12 12c-.578 0-1.065.21-1.407.56-.34.346-.593.887-.593 1.645H8V14h.004c.044-1.138.457-2.124 1.16-2.841C9.909 10.4 10.922 10 12 10Zm-5 7v2h10v-2H7Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      <div
        class="c8 c9"
      />
      <div
        class="c10 c11"
        id="w-overview-PersonBody--element-nationality"
      >
        <p
          class="navds-detail"
        >
          <strong>
            ui:nationality
          </strong>
          :
        </p>
        <div
          class="c8 c9"
        />
        <div
          class="navds-body-long"
        >
          Norge (09.02.1980)
        </div>
      </div>
    </div>
  </div>
  <div
    class="c18 c19"
  />
  <div
    class="c0 c1 c2 c3"
  >
    <div
      class="c4 c5 c6 c7"
    >
      <svg
        fill="none"
        focusable="false"
        height="1em"
        role="img"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M12 0c6.627 0 12 5.373 12 12 0 1.707-.356 3.33-.999 4.801L23 17l-.089.001C21.015 21.131 16.842 24 12 24c-4.843 0-9.015-2.868-10.911-6.999L1 17v-.196A11.96 11.96 0 0 1 0 12C0 5.373 5.373 0 12 0Zm2.503 17.001H9.497C10.12 20.016 11.207 22 12 22c.793 0 1.88-1.984 2.503-4.999Zm-7.05 0H3.339a10.028 10.028 0 0 0 5.623 4.529c-.644-1.185-1.165-2.738-1.507-4.53Zm13.209 0h-4.115c-.343 1.79-.864 3.344-1.508 4.53A10.03 10.03 0 0 0 20.662 17Zm-13.593-7L2.2 10a10.046 10.046 0 0 0 .258 5h4.7a28.7 28.7 0 0 1-.088-5Zm7.854 0H9.077a25.939 25.939 0 0 0 .098 5h5.65a25.716 25.716 0 0 0 .098-5ZM21.8 10h-4.87a28.732 28.732 0 0 1-.088 5h4.7a10.052 10.052 0 0 0 .258-5ZM8.961 2.47l-.172.057A10.03 10.03 0 0 0 2.832 8h4.452c.305-2.067.838-3.873 1.527-5.244l.15-.287ZM12 2c-.878 0-2.115 2.43-2.687 6h5.374c-.557-3.473-1.743-5.869-2.616-5.995L12 2Zm3.039.47.043.08c.74 1.394 1.314 3.279 1.633 5.45h4.453a10.029 10.029 0 0 0-6.13-5.53Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      <div
        class="c8 c9"
      />
      <div
        class="c10 c11"
        id="w-overview-PersonBody--element-oppholdsadresse"
      >
        <p
          class="navds-detail"
        >
          <strong>
            ui:oppholdsadresse
          </strong>
          :
        </p>
        <div
          class="c8 c9"
        />
        <div
          class="navds-body-long"
        >
          <div
            aria-hidden="true"
            class="navds-popover navds-popover--hidden"
            data-placement="right"
            style="position: absolute; top: 0px; left: 0px;"
            tabindex="-1"
          >
            <div
              class="c12 c13"
            >
              <span>
                ui:fram-og-til
              </span>
            </div>
            <div
              class="navds-popover__arrow"
              style="left: -0.5rem;"
            />
          </div>
          <div
            class="c14 c15"
            id="15e756c8-6070-4958-91da-c3c32a947b1b"
          >
            <span>
              01.01.2022 - 01.01.2023
            </span>
          </div>
          <br />
          <div
            aria-hidden="true"
            class="navds-popover navds-popover--hidden"
            data-placement="right"
            style="position: absolute; top: 0px; left: 0px;"
            tabindex="-1"
          >
            <div
              class="c12 c13"
            >
              <span>
                ui:adressenavn
              </span>
            </div>
            <div
              class="navds-popover__arrow"
              style="left: -0.5rem;"
            />
          </div>
          <div
            class="c14 c15"
            id="b1dab881-b1b4-4729-8bac-0d85251ae9dd"
          >
            <span>
              Adressenavn2
            </span>
          </div>
          <span
            class="c16 c17"
          />
          <div
            aria-hidden="true"
            class="navds-popover navds-popover--hidden"
            data-placement="right"
            style="position: absolute; top: 0px; left: 0px;"
            tabindex="-1"
          >
            <div
              class="c12 c13"
            >
              <span>
                ui:husnummer
              </span>
            </div>
            <div
              class="navds-popover__arrow"
              style="left: -0.5rem;"
            />
          </div>
          <div
            class="c14 c15"
            id="f0101045-d1b8-4caa-bc16-d9713f90db4a"
          >
            <span>
              01
            </span>
          </div>
          <span
            class="c16 c17"
          />
          <div
            aria-hidden="true"
            class="navds-popover navds-popover--hidden"
            data-placement="right"
            style="position: absolute; top: 0px; left: 0px;"
            tabindex="-1"
          >
            <div
              class="c12 c13"
            >
              <span>
                ui:husbokstav
              </span>
            </div>
            <div
              class="navds-popover__arrow"
              style="left: -0.5rem;"
            />
          </div>
          <div
            class="c14 c15"
            id="2d098ade-462b-486b-8c00-a3b1c248ccc2"
          >
            <span>
              B
            </span>
          </div>
          <br />
          <div
            aria-hidden="true"
            class="navds-popover navds-popover--hidden"
            data-placement="right"
            style="position: absolute; top: 0px; left: 0px;"
            tabindex="-1"
          >
            <div
              class="c12 c13"
            >
              <span>
                ui:poststed
              </span>
            </div>
            <div
              class="navds-popover__arrow"
              style="left: -0.5rem;"
            />
          </div>
          <div
            class="c14 c15"
            id="c172473e-5a4d-44a9-a780-a22e04a08d21"
          >
            <span>
              0768
            </span>
          </div>
          <span
            class="c16 c17"
          />
          <div
            aria-hidden="true"
            class="navds-popover navds-popover--hidden"
            data-placement="right"
            style="position: absolute; top: 0px; left: 0px;"
            tabindex="-1"
          >
            <div
              class="c12 c13"
            >
              <span>
                ui:city
              </span>
            </div>
            <div
              class="navds-popover__arrow"
              style="left: -0.5rem;"
            />
          </div>
          <div
            class="c14 c15"
            id="d6a104a6-fc6a-4765-b62b-d21f9cd58695"
          >
            <span>
              OSLO
            </span>
          </div>
          <br />
        </div>
      </div>
    </div>
    <div
      class="c4 c5 c6 c7"
    >
      <svg
        fill="none"
        focusable="false"
        height="1em"
        role="img"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M17 1c3.866 0 7 3.283 7 7.333a7.486 7.486 0 0 1-2.103 5.24l.103-.002L12 23 2 13.571l.103.002A7.486 7.486 0 0 1 0 8.333C0 4.283 3.134 1 7 1c1.959 0 3.73.843 5 2.202C13.27 1.843 15.042 1 17 1Zm0 2c-1.26 0-2.445.498-3.359 1.385l-.179.182-1.46 1.564-1.462-1.563C9.603 3.566 8.344 3 7 3 4.26 3 2 5.368 2 8.333c0 1.383.494 2.674 1.357 3.653l.178.191 1.102 1.131L12 20.25l7.362-6.942 1.103-1.13a5.469 5.469 0 0 0 1.529-3.57L22 8.334C22 5.368 19.74 3 17 3Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      <div
        class="c8 c9"
      />
      <div
        class="c10 c11"
        id="w-overview-PersonBody--element-marital-status"
      >
        <p
          class="navds-detail"
        >
          <strong>
            ui:marital-status
          </strong>
          :
        </p>
        <div
          class="c8 c9"
        />
        <div
          class="navds-body-long"
        >
          Gift (10.10.2007)
        </div>
      </div>
    </div>
    <div
      class="c4 c5 c6 c7"
    />
  </div>
  <div
    class="c18 c19"
  />
  <div
    class="c0 c1 c2 c3"
  >
    <div
      class="c4 c5 c6 c7"
    >
      <svg
        fill="none"
        focusable="false"
        height="1em"
        role="img"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M9 7a3 3 0 1 1 6 0 3 3 0 0 1-6 0Zm3-5a5 5 0 1 0 0 10 5 5 0 0 0 0-10ZM6 21a6 6 0 0 1 12 0v1h2v-1a8 8 0 1 0-16 0v1h2v-1Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      <div
        class="c8 c9"
      />
      <p
        class="navds-detail navds-detail--small"
      >
        <strong>
          ui:deceased
        </strong>
        :
      </p>
      <div>
        <div
          class="c10 c11"
          id="w-overview-PersonBody--element-deceased"
        >
          <div
            class="c8 c9"
          />
          <p
            class="navds-body-long"
          >
            MOR AVDØD1 - personFarFnr (buc:relasjon-MOR)
          </p>
        </div>
      </div>
    </div>
    <div
      class="c4 c5 c6 c7"
    />
  </div>
</div>
`;

exports[`applications/PersonPanel/PersonBody Render: returns empty with no person 1`] = `null`;
